pkg/dao/templates.go

```diff
@@ -69,6 +69,14 @@ func (t templateDaoImpl) create(ctx context.Context, tx *gorm.DB, reqTemplate ap
 		return api.TemplateResponse{}, err
 	}
 
+	// Ensure arch and version exist and are not blank
+	if reqTemplate.Arch == nil || reqTemplate.Version == nil || *reqTemplate.Arch == "" || *reqTemplate.Version == "" {
+		return api.TemplateResponse{}, &ce.DaoError{
+			Message:       "template must include arch and version",
+			BadValidation: true,
+		}
+	}
```

Comment:
It might make sense to check for this in the models package, where we already have some validation for arch and version. What do you think?

Reply:
oh yep that's a much better place :) will fix

Reply:
should i also probably remove the handler and dao tests here? and just add a test in the models pkg?

Reply:
i agree, good call!

Reply:
ok added some test cases to the model. thanks for pointing that out, that was silly of me :)